backend/etl/tests/test_data_handler.py

```diff
@@ -205,11 +206,24 @@ def test_fetch_data_request_exception(data_handler, caplog):
     assert "API Error" in caplog.text
 
 
+@responses.activate
 def test_fetch_data_retry_exhausted(fast_retry_session, caplog):
     """Test that retry mechanism works and eventually exhausts"""
+
+    fake_url = "https://fake.test/504"
+
+    # Simulate 6 (1 initial attempt and 5 retries) consecutive 504 responses
+    for _ in range(6):
+        responses.add(
+            responses.GET,
+            fake_url,
+            status=504,
+            json={},
+        )
+
     # Create handler with fast retry session
     data_handler = DummyDataHandler(
-        url="https://httpstat.us/504",
+        url=fake_url,
```

Comment:
Line 237 reads "Retry Attempt 3 of 5"; I believe that means the assertion is looking for this string. Just double checking that this is deliberate (versus, for example, "Retry Attempt 5 of 5" or similar).